modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -596,6 +643,212 @@ private QueryPlan prepareDml(SqlNode sqlNode, PlanningContext ctx) throws Valida
+                            if (!(opt.value() instanceof SqlIdentifier))
+                                throw new IgniteSQLException("Unexpected value for param " + VALUE_TYPE + " [" +
+                                    "expected an identifier, but was " + opt.value() + "; " +
+                                    "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.PARSING);
+
+                            createTblCmd.valueTypeName(((SqlIdentifier)opt.value()).names.get(0));
+                            break;
+
+                        case WRITE_SYNCHRONIZATION_MODE: {
+                            CacheWriteSynchronizationMode mode = null;
+
+                            if (opt.value() instanceof SqlIdentifier) {
+                                mode = Arrays.stream(CacheWriteSynchronizationMode.values())
+                                    .filter(m -> m.name().equalsIgnoreCase(opt.value().toString()))
+                                    .findFirst()
+                                    .orElse(null);
+                            }
+
+                            if (mode == null)
+                                throw new IgniteSQLException("Unexpected value for param " + WRITE_SYNCHRONIZATION_MODE
+                                    + " [expected values are " + Arrays.toString(CacheWriteSynchronizationMode.values())
+                                    + ", but was " + opt.value() + "; " +
+                                    "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.PARSING);
+
+                            createTblCmd.writeSynchronizationMode(mode);
+                            break;
+                        }
+                        case ENCRYPTED:
+                            if (!(opt.value() instanceof SqlLiteral) && ((SqlLiteral)opt.value()).getTypeName() != BOOLEAN)
+                                throw new IgniteSQLException("Unexpected value for param " + ENCRYPTED + " [" +
+                                    "expected a boolean, but was " + opt.value() + "; " +
+                                    "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.PARSING);
+
+                            createTblCmd.encrypted(((SqlLiteral)opt.value()).booleanValue());
+                            break;
+
+                        default:
+                            throw new IllegalStateException("Unsupported option " + opt.key());
+                    }
+                }
+            }
+
+            List<SqlColumnDeclaration> colDeclarations = createTblNode.columnList().getList().stream()
+                .filter(SqlColumnDeclaration.class::isInstance)
+                .map(SqlColumnDeclaration.class::cast)
+                .collect(Collectors.toList());
+
+            IgnitePlanner planner = ctx.planner();
+
+            List<ColumnDefinition> cols = new ArrayList<>();
+
+            for (SqlColumnDeclaration col : colDeclarations) {
+                String name = Util.last(col.name.names);
```

Comment:
we are going to support a compound identifier? As I remember 'No'. If so, maybe better to check that we have a simple name?

Reply:
fixed